app/models/user.rb

```diff
@@ -19,4 +30,12 @@ def self.find_for_github_oauth(auth)
   def self.create_unique_string
     SecureRandom.uuid
   end
+
+  def followed_by?(user)
+    self.passive_friendships.find_by(following_id: user.id).present?
+  end
+
+  def self.timeline(user)
+    Book.where("user_id IN (?) OR user_id = ?", user.following_ids, user.id).order(created_at: :desc)
```

Comment:
```suggestion
    Book.where(user_id: user.following_ids + [user.id]).order(created_at: :desc)
```

もしくは`User`モデルのscopeとして作れば引数のいらないscopeになりそうです。